identity-apps-core/components/org.wso2.identity.apps.common/src/main/java/org/wso2/identity/apps/common/listner/AppPortalOAuthAppMgtListener.java

```diff
@@ -64,7 +65,8 @@ public void doPreUpdateConsumerApplication(OAuthConsumerAppDTO oAuthConsumerAppD
             throws IdentityOAuthAdminException {
 
         if (!isEnabled() || systemAppConsumerKeys.stream()
-                .noneMatch(oAuthConsumerAppDTO.getApplicationName()::equalsIgnoreCase)) {
+                .noneMatch(oAuthConsumerAppDTO.getApplicationName()::equalsIgnoreCase) ||
+            IdentityApplicationManagementUtil.getAllowUpdateSystemApplicationThreadLocal()) {
```

Comment:
Instead of using a thread local here, can't we allow changes if a system appliaction's callback/allowed origin has changed?

Reply:
We can not do the validation here since here we got the `OAuthConsumerAppDTO` object instead of `OpenIDConnectConfiguration` object. This `OAuthConsumerAppDTO` doesn't contains all the attributes includes in `OpenIDConnectConfiguration` object. So we cannot compare existing OpenIDConnectConfiguration configurations with newly added OpenIDConnectConfiguration configurations here.